src/index.ts

```diff
@@ -415,6 +432,87 @@ export class FlashbotsBundleProvider extends providers.JsonRpcProvider {
     }
   }
 
+  /**
+   * Method to send a carrier tx into the public mempool
+   *
+   * @param bundle  FlashbotsBundle with AT LEAST signed bundled transactions in signedBundledTransactions field obtained
+   *  from {@link signBundle} method, and blockTarget.
+   * @param validatorPublicKey  The public key of the validator that will be able to decrypt the bundle and include it
+   *  into the bundle pool.
+   * @param signer  Signer who will sign the carrier transaction.
+   * @param carrierTx TransactionRequest whose data field will carry the encrypted bundle : MAY be an incomplete
+   *  object which will be populated with default values.
+   *
+   * @return Promise<TransactionResponse> Promise containing the response for the carrier tx
+   * */
+
+  public async publish(
+    bundle: FlashbotsBundle,
+    validatorPublicKey: string,
+    signer: Signer,
+    carrierTx: TransactionRequest
+  ): Promise<TransactionResponse> {
+    //RLP-serialize the given bundle
+    const serializedBundle = this.rlpSerializeBundle(bundle)
+
+    //Encrypt the encoded bundle with the passed validator pub_key
+    const encryptedBundle = encrypt(validatorPublicKey, Buffer.from(serializedBundle))
```

Comment:
perhaps just serialize and sign an array of txs instead of a whole bundle? (reasons given in struct above)

Reply:
seen and responded above